4 feature

TomaszSarmata opened the pull request:
added another endpoint with all relevant tests

TomaszSarmata committed:
added /api/articles/:article_id along with relevant tests

TomaszSarmata committed:
changed the error handling middleware chain to follow the coursework / notes

TomaszSarmata committed:
code clean-up only

nc-rayhaan commented on __tests__/api.test.js:
```diff
@@ -65,3 +64,40 @@ describe("GET: /api", () => {
+          created_at: expect.any(String),
+          votes: expect.any(Number),
+          article_img_url: expect.any(String),
+        });
```
As we are only receiving one object, we can explicitly test its values instead of using `expect.any()`

nc-rayhaan commented on __tests__/api.test.js:
```diff
@@ -65,3 +64,40 @@ describe("GET: /api", () => {
+      .expect(404)
+      .then((res) => {
+        expect(res.body.msg).toBe("Bad Request");
+      });
```
The status code is correct, however, your message implies this is a bad request. This is an article that doesn't exist so wouldn't be a bad request. Please do change the message

nc-rayhaan commented on models/api.models.js:
```diff
@@ -16,9 +17,29 @@ const fetchAllTopics = () => {
+const fetchArticleById = (id) => {
+  if (id === undefined || !typeof id === "number") {
+    return Promise.reject({ status: 400, msg: "Bad Request" });
+  }
```
There is no need to do this if statement. If the id is not a number, PSQL will throw an error and you can handle it that way. The id would never undefined as that would be a different endpoint altogether

nc-rayhaan reviewed (commented):
Nice work overall! I've left some comments for you. Please do fix these when you can 🙂

TomaszSarmata commented:
Thanks very much Rayhaan. I've just removed the unnecessary sanity check from fetchArticleById and also changed the message in the 404 for the same endpoint to be a bit more clear with the reason for the error.
